src/gromacs/nblib/molecules.cpp

```diff
@@ -45,9 +45,25 @@ Molecule& Molecule::addAtom(const std::string &atomName, const std::string &resi
     return *this;
 }
 
-Molecule& Molecule::addAtom(const std::string &atomName, AtomType const &atomType)
+Molecule& Molecule::addAtom(const AtomName& atomName, const ResidueName& residueName, AtomType const &atomType)
 {
-    addAtom(atomName, name_, atomType);
+    real charge = 0;
```

Comment:
Do you think an order-agnostic, named argument list approach would work better here?
```
addAtom(AtomName("Oc"), ResidueName("Carboxyl"), Charge(-0.6), AtomType(oc));
```

Reply:
I did not try to implement that but it could be nice. The problem I suspect will come in overload resolution. If you have a quick solution here I would say it is fine to push it. However, I am not sure that it would add enough convenience for it to be worth spending much time to get it working.

Reply:
I agree with Joe that for now it doesn't seem worth the effort. We can of course come back to this if the issue arises in other parts of the code.